dces-drc-integration/src/main/java/uk/gov/justice/laa/crime/dces/integration/config/DrcApiWebClientConfiguration.java

```diff
@@ -50,13 +65,51 @@ public WebClient drcApiWebClient(
                 .baseUrl(servicesConfiguration.getDrcClientApi().getBaseUrl())
                 .clientConnector(
                         new ReactorClientHttpConnector(
-                                HttpClient.create(provider)
-                                        .resolver(DefaultAddressResolverGroup.INSTANCE)
-                                        .compress(true)
-                                        .responseTimeout(Duration.ofSeconds(30))))
+                                createHttpClient(provider, sslBundles)))
                 .build();
     }
 
+    private static HttpClient createHttpClient(ConnectionProvider provider, SslBundles sslBundles) {
+        HttpClient httpClient = HttpClient.create(provider);
+        // Handle client-authentication.
+        final SslBundle clientAuthBundle = getClientAuthBundle(sslBundles);
+        if (clientAuthBundle != null) {
+            httpClient = httpClient.secure(spec -> {
+                SslOptions options = clientAuthBundle.getOptions();
+                SslManagerBundle managers = clientAuthBundle.getManagers();
+                SslContextBuilder builder = SslContextBuilder.forClient()
+                        .keyManager(managers.getKeyManagerFactory())
+                        .trustManager(managers.getTrustManagerFactory())
+                        .ciphers(SslOptions.asSet(options.getCiphers()))
+                        .protocols(options.getEnabledProtocols());
+                try {
+                    spec.sslContext(builder.build());
+                } catch (SSLException e) {
+                    throw new IllegalStateException(e);
+                }
+            });
+        }
+        return httpClient
+                .resolver(DefaultAddressResolverGroup.INSTANCE)
+                .compress(true)
+                .responseTimeout(Duration.ofSeconds(30));
+    }
+
+    private static SslBundle getClientAuthBundle(SslBundles sslBundles) {
+        SslBundle sslBundle = null;
+        if (sslBundles != null) {
+            try {
+                sslBundle = sslBundles.getBundle("client-auth");
+                log.info("SSL Bundle 'client-auth' retrieved: {}", sslBundle);
+            } catch (NoSuchSslBundleException e) {
+                log.info("SSL Bundle 'client-auth' not available: {}", e.getMessage());
+            }
+        } else {
+            log.info("SSL Bundles not available: null");
+        }
+        return sslBundle;
+    }
+
     @Bean
     DrcClient crimeApplyDatastoreClient(WebClient drcApiWebClient) {
```

Comment:
This is not your change and probably an older copy-paste problem, but can we rename this method, to avoid confusion in future?

Reply:
yep - can do. I actually originally did - but wanted to keep the change simple (it has a knock-on effect on a test name too).